src/transport_udp.cpp

```diff
@@ -258,8 +258,8 @@ UDPTransport::fd_reader()
                         &remoteAddrLen);
 
     if (rLen < 0) {
-      if (errno == EAGAIN) {
-        // timeout on read
+      if ((errno == EAGAIN) || (stop)) {
```

Comment:
I also ran into the reader thread not closing nicely due to FD being closed, but adding ```errno == EBADF``` sufficed.  This is needed anyway for use cases that the FD is closed by other means. Difference is that we don't want to ```continue```, instead ```break```.

Reply:
I just put the check for stop() here so that processing would not continue on stop.  I didn't try to address all possible failure scenarios.  I do wonder why we use this as opposed to the Cantina::Network object.  It has proven to be pretty robust, as we're using that in Gallia.  It can use one or more threads to deliver data packets asynchronously to the application.  I also used it in QoH3.